sql/database.go

```diff
@@ -176,6 +184,21 @@ func withForceFresh() Opt {
 	}
 }
 
+// WithTemp specifies temporary database mode.
+// For the temporary database, the migrations are always run in place, and vacuuming is
+// nover done.  PRAGMA journal_mode=OFF and PRAGMA synchronous=OFF are used.
+func WithTemp() Opt {
```

Comment:
Not sure how to approach improving this within the scope of this PR. It would probably produce a bit too many different funcs for doing mostly the same (opening the DB), given that we already have a few different ones for that. Probably a subject for further discussion...

Reply:
I'm not sure if this specific functional option is incorrect.

By default we do not want to run migrations in place and we do want to vacuum the DB, so one can add `sql.WithTemp()` to disable this behaviour (for specific cases, like in tests).